test/api/unit/services/GithubBuildStatusReporter.test.js

```diff
@@ -0,0 +1,218 @@
+const crypto = require("crypto")
+const expect = require("chai").expect
+const nock = require("nock")
+const factory = require("../../support/factory")
+const githubAPINocks = require("../../support/githubAPINocks")
+const { Build, User } = require("../../../../api/models")
+
+const GithubBuildStatusReporter = require("../../../../api/services/GithubBuildStatusReporter")
+
+describe("GithubBuildStatusReporter", () => {
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  describe("reportBuildStatus(build)", () => {
+    context("with a build in the processing state", () => {
+      it("should report that the status is 'pending'", done => {
+        let statusNock
+
+        factory.build({
+          state: "processing",
+          site: factory.site({ owner: "test-owner", repository: "test-repo" }),
+          commitSha: "456def",
+        }).then(build => {
+          statusNock = githubAPINocks.status({
+            owner: "test-owner",
+            repo: "test-repo",
+            repository: "test-repo",
+            sha: "456def",
+            state: "pending",
+          })
+
+          return GithubBuildStatusReporter.reportBuildStatus(build)
+        }).then(() => {
+          expect(statusNock.isDone()).to.be.true
+          done()
+        }).catch(done)
+      })
+
+      it("should set the target uri to the build logs", done => {
+        let statusNock
+
+        factory.build({
+          state: "processing",
+          site: factory.site({ owner: "test-owner", repository: "test-repo" }),
+          commitSha: "456def",
+        }).then(build => {
+          statusNock = githubAPINocks.status({
+            owner: "test-owner",
+            repo: "test-repo",
+            sha: "456def",
+            targetURL: `http://localhost:1337/sites/${build.site}/builds/${build.id}/logs`,
```

Comment:
We're coupling our expected host/port/scheme combination with strings here. Not critical, but a string change somewhere else can break these tests without actually breaking this functionality.